Index: configure
===================================================================
RCS file: /cvsroot/clisp/clisp/src/configure,v
retrieving revision 1.81
retrieving revision 1.82
diff -u -w -b -u -b -w -i -B -r1.81 -r1.82
--- configure	4 Sep 2003 21:33:23 -0000	1.81
+++ configure	5 Sep 2003 14:41:30 -0000	1.82
@@ -14570,26 +14570,6 @@
 #endif
 /* Declare opendir(), closedir(). */
 #include <$ac_header_dirent>
-extern
-#ifdef __cplusplus
-"C"
-#endif
-
-#if defined(__STDC__) || defined(__cplusplus)
-DIR* opendir (OPENDIR_CONST char* dirname);
-#else
-DIR* opendir();
-#endif
-extern
-#ifdef __cplusplus
-"C"
-#endif
-
-#if defined(__STDC__) || defined(__cplusplus)
-RETCLOSEDIRTYPE closedir (DIR* dirp);
-#else
-RETCLOSEDIRTYPE closedir();
-#endif
 int main() { exit(closedir(opendir(".")) != 0); }
 _ACEOF
 rm -f conftest$ac_exeext
@@ -15526,31 +15506,11 @@
 #ifdef NEED_SYS_IOCTL_H
 #include <sys/ioctl.h>
 #endif
-extern
-#ifdef __cplusplus
-"C"
-#endif
-
-#if defined(__STDC__) || defined(__cplusplus)
-int ioctl ($cl_cv_proto_ioctl_args);
-#else
-int ioctl();
-#endif
 /* Declare open(). */
 #include <fcntl.h>
 #ifdef OPEN_NEEDS_SYS_FILE_H
 #include <sys/file.h>
 #endif
-extern
-#ifdef __cplusplus
-"C"
-#endif
-
-#if defined(__STDC__) || defined(__cplusplus)
-int open ($cl_cv_proto_open_args);
-#else
-int open();
-#endif
 int main ()
 { int fd = open("conftest.c",O_RDONLY,0644);
   long x;
@@ -17783,18 +17743,6 @@
 #ifdef HAVE_UNISTD_H
 #include <unistd.h>
 #endif
-#ifndef malloc
-extern
-#ifdef __cplusplus
-"C"
-#endif
-
-#if defined(__STDC__) || defined(__cplusplus)
-RETMALLOCTYPE malloc (MALLOC_SIZE_T size);
-#else
-RETMALLOCTYPE malloc();
-#endif
-#endif
 $address_range_prog
 int main() { printf_address(chop_address(malloc(10000))); exit(0); }
 EOF
@@ -19250,44 +19198,11 @@
 #ifdef HAVE_UNISTD_H
 #include <unistd.h>
 #endif
-#ifndef malloc
-extern
-#ifdef __cplusplus
-"C"
-#endif
-
-#if defined(__STDC__) || defined(__cplusplus)
-RETMALLOCTYPE malloc (MALLOC_SIZE_T size);
-#else
-RETMALLOCTYPE malloc();
-#endif
-#endif
 /* declare getpagesize() and mprotect() */
 #include <sys/mman.h>
 #ifndef HAVE_GETPAGESIZE
 #include <sys/param.h>
 #define getpagesize() PAGESIZE
-#else
-extern
-#ifdef __cplusplus
-"C"
-#endif
-
-#if defined(__STDC__) || defined(__cplusplus)
-RETGETPAGESIZETYPE getpagesize (void);
-#else
-RETGETPAGESIZETYPE getpagesize();
-#endif
-#endif
-extern
-#ifdef __cplusplus
-"C"
-#endif
-
-#if defined(__STDC__) || defined(__cplusplus)
-int mprotect (MPROTECT_CONST MMAP_ADDR_T addr, MMAP_SIZE_T len, int prot);
-#else
-int mprotect();
 #endif
 char foo;
 int main () {
@@ -19516,16 +19431,12 @@
 fi
 
 if test "$ac_cv_header_sys_shm_h" = yes -a "$ac_cv_header_sys_ipc_h" = yes; then
-# This test is from Marcus Daniels
-echo "$as_me:$LINENO: checking for working shared memory" >&5
-echo $ECHO_N "checking for working shared memory... $ECHO_C" >&6
-if test "${cl_cv_sys_shm_works+set}" = set; then
+echo "$as_me:$LINENO: checking for shmat declaration" >&5
+echo $ECHO_N "checking for shmat declaration... $ECHO_C" >&6
+if test "${cl_cv_proto_shmat+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
 
-if test "$cross_compiling" = yes; then
-  cl_cv_sys_shm_works="guessing no"
-else
   cat >conftest.$ac_ext <<_ACEOF
 #line $LINENO "configure"
 /* confdefs.h.  */
@@ -19533,6 +19444,11 @@
 cat confdefs.h >>conftest.$ac_ext
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
+
+#include <stdlib.h>
+#ifdef HAVE_UNISTD_H
+#include <unistd.h>
+#endif
 #include <sys/types.h>
 #include <sys/ipc.h>
 #include <sys/shm.h>
@@ -19542,40 +19459,146 @@
 #endif
 
 #if defined(__STDC__) || defined(__cplusplus)
-int shmget (key_t key, $cl_cv_proto_shmget_arg2 size, int shmflg);
-#else
-int shmget();
-#endif
-extern
+
 #ifdef __cplusplus
-"C"
+void* shmat(int, const void *, int);
+#else
+void* shmat();
 #endif
 
-#if defined(__STDC__) || defined(__cplusplus)
-$cl_cv_proto_shmat_ret shmat (int shmid, $cl_cv_proto_shmat_arg2 $cl_cv_proto_shmat_ret shmaddr, int shmflg);
 #else
-$cl_cv_proto_shmat_ret shmat();
-#endif
-extern
-#ifdef __cplusplus
-"C"
+void* shmat();
 #endif
 
-#if defined(__STDC__) || defined(__cplusplus)
-int shmdt ($cl_cv_proto_shmdt_arg1 shmaddr);
-#else
-int shmdt();
+int
+main ()
+{
+
+  ;
+  return 0;
+}
+_ACEOF
+rm -f conftest.$ac_objext
+if { (eval echo "$as_me:$LINENO: \"$ac_compile\"") >&5
+  (eval $ac_compile) 2>&5
+  ac_status=$?
+  echo "$as_me:$LINENO: \$? = $ac_status" >&5
+  (exit $ac_status); } &&
+         { ac_try='test -s conftest.$ac_objext'
+  { (eval echo "$as_me:$LINENO: \"$ac_try\"") >&5
+  (eval $ac_try) 2>&5
+  ac_status=$?
+  echo "$as_me:$LINENO: \$? = $ac_status" >&5
+  (exit $ac_status); }; }; then
+  cl_cv_proto_shmat_ret="void*"
+else
+  echo "$as_me: failed program was:" >&5
+sed 's/^/| /' conftest.$ac_ext >&5
+
+cl_cv_proto_shmat_ret="char*"
+fi
+rm -f conftest.$ac_objext conftest.$ac_ext
+
+cat >conftest.$ac_ext <<_ACEOF
+#line $LINENO "configure"
+/* confdefs.h.  */
+_ACEOF
+cat confdefs.h >>conftest.$ac_ext
+cat >>conftest.$ac_ext <<_ACEOF
+/* end confdefs.h.  */
+
+#include <stdlib.h>
+#ifdef HAVE_UNISTD_H
+#include <unistd.h>
 #endif
+#include <sys/types.h>
+#include <sys/ipc.h>
+#include <sys/shm.h>
+
 extern
 #ifdef __cplusplus
 "C"
 #endif
 
 #if defined(__STDC__) || defined(__cplusplus)
-int shmctl ($cl_cv_proto_shmctl_args);
+$cl_cv_proto_shmat_ret shmat (int shmid, $cl_cv_proto_shmat_ret shmaddr, int shmflg);
 #else
-int shmctl();
+$cl_cv_proto_shmat_ret shmat();
 #endif
+
+int
+main ()
+{
+
+  ;
+  return 0;
+}
+_ACEOF
+rm -f conftest.$ac_objext
+if { (eval echo "$as_me:$LINENO: \"$ac_compile\"") >&5
+  (eval $ac_compile) 2>&5
+  ac_status=$?
+  echo "$as_me:$LINENO: \$? = $ac_status" >&5
+  (exit $ac_status); } &&
+         { ac_try='test -s conftest.$ac_objext'
+  { (eval echo "$as_me:$LINENO: \"$ac_try\"") >&5
+  (eval $ac_try) 2>&5
+  ac_status=$?
+  echo "$as_me:$LINENO: \$? = $ac_status" >&5
+  (exit $ac_status); }; }; then
+  cl_cv_proto_shmat_arg2=""
+else
+  echo "$as_me: failed program was:" >&5
+sed 's/^/| /' conftest.$ac_ext >&5
+
+cl_cv_proto_shmat_arg2="const"
+fi
+rm -f conftest.$ac_objext conftest.$ac_ext
+
+
+
+cl_cv_proto_shmat="extern $cl_cv_proto_shmat_ret shmat (int, $cl_cv_proto_shmat_arg2 $cl_cv_proto_shmat_ret, int);"
+fi
+
+cl_cv_proto_shmat=`echo "$cl_cv_proto_shmat" | tr -s ' ' | sed -e 's/( /(/'`
+echo "$as_me:$LINENO: result: ${ac_t:-
+         }$cl_cv_proto_shmat" >&5
+echo "${ECHO_T}${ac_t:-
+         }$cl_cv_proto_shmat" >&6
+
+
+cat >>confdefs.h <<_ACEOF
+#define RETSHMATTYPE $cl_cv_proto_shmat_ret
+_ACEOF
+
+
+cat >>confdefs.h <<_ACEOF
+#define SHMAT_CONST $cl_cv_proto_shmat_arg2
+_ACEOF
+
+fi
+
+if test "$ac_cv_header_sys_shm_h" = yes -a "$ac_cv_header_sys_ipc_h" = yes; then
+# This test is from Marcus Daniels
+echo "$as_me:$LINENO: checking for working shared memory" >&5
+echo $ECHO_N "checking for working shared memory... $ECHO_C" >&6
+if test "${cl_cv_sys_shm_works+set}" = set; then
+  echo $ECHO_N "(cached) $ECHO_C" >&6
+else
+
+if test "$cross_compiling" = yes; then
+  cl_cv_sys_shm_works="guessing no"
+else
+  cat >conftest.$ac_ext <<_ACEOF
+#line $LINENO "configure"
+/* confdefs.h.  */
+_ACEOF
+cat confdefs.h >>conftest.$ac_ext
+cat >>conftest.$ac_ext <<_ACEOF
+/* end confdefs.h.  */
+#include <sys/types.h>
+#include <sys/ipc.h>
+#include <sys/shm.h>
 /* try attaching a single segment to multiple addresses */
 #define segsize 0x10000
 #define attaches 128
@@ -19799,32 +19822,6 @@
 #ifdef HAVE_SYS_SYSMACROS_H
 #include <sys/sysmacros.h>
 #endif
-#if defined(__STDC__) || defined(__cplusplus)
-extern
-#ifdef __cplusplus
-"C"
-#endif
-int shmget (key_t key, SHMGET_SIZE_T size, int shmflg);
-extern
-#ifdef __cplusplus
-"C"
-#endif
-RETSHMATTYPE shmat (int shmid, SHMAT_CONST RETSHMATTYPE shmaddr, int shmflg);
-extern
-#ifdef __cplusplus
-"C"
-#endif
-
-#ifdef SHMCTL_DOTS
-int shmctl (int shmid, int cmd, ...);
-#else
-int shmctl (int shmid, int cmd, struct shmid_ds * buf);
-#endif
-#else
-extern int shmget();
-extern RETSHMATTYPE shmat();
-extern int shmctl();
-#endif
 int main ()
 { unsigned int pagesize = 8192; /* should be a multiple of SHMLBA */
   unsigned long addr = (unsigned long) malloc(2*pagesize);
@@ -21214,37 +21211,17 @@
 #include "confdefs.h"
 #include <sys/types.h>
 #include <stdlib.h>
+/* Declare chdir(). */
 #ifdef HAVE_UNISTD_H
 #include <unistd.h>
 #endif
 #include <string.h>
 #include <stdio.h>
-/* Declare chdir(). */
-extern
-#ifdef __cplusplus
-"C"
-#endif
-
-#if defined(__STDC__) || defined(__cplusplus)
-int chdir (CHDIR_CONST char* path);
-#else
-int chdir();
-#endif
 /* Declare open(). */
 #include <fcntl.h>
 #ifdef OPEN_NEEDS_SYS_FILE_H
 #include <sys/file.h>
 #endif
-extern
-#ifdef __cplusplus
-"C"
-#endif
-
-#if defined(__STDC__) || defined(__cplusplus)
-int open ($cl_cv_proto_open_args);
-#else
-int open();
-#endif
 /* Declare opendir(), readdir(), closedir(). */
 #include <$ac_header_dirent>
 #ifdef DIRENT
@@ -21252,36 +21229,6 @@
 #else
 #define SDIRENT struct direct
 #endif
-extern
-#ifdef __cplusplus
-"C"
-#endif
-
-#if defined(__STDC__) || defined(__cplusplus)
-DIR* opendir (OPENDIR_CONST char* dirname);
-#else
-DIR* opendir();
-#endif
-extern
-#ifdef __cplusplus
-"C"
-#endif
-
-#if defined(__STDC__) || defined(__cplusplus)
-SDIRENT* readdir (DIR* dirp);
-#else
-SDIRENT* readdir();
-#endif
-extern
-#ifdef __cplusplus
-"C"
-#endif
-
-#if defined(__STDC__) || defined(__cplusplus)
-RETCLOSEDIRTYPE closedir (DIR* dirp);
-#else
-RETCLOSEDIRTYPE closedir();
-#endif
 /* A small program which checks for each character whether or not it is
  * valid in filenames. */
 #define N 256
@@ -21397,7 +21344,7 @@
 echo "${ECHO_T}$cl_cv_os_valid_filename_charset" >&6
 if test -n "$cl_cv_os_valid_filename_char"; then
 
-cat >>confdefs.h <<\_ACEOF
+cat >>confdefs.h <<_ACEOF
 #define VALID_FILENAME_CHAR $cl_cv_os_valid_filename_char
 _ACEOF
 

