--- clisp-2.49/modules/bindings/glibc/linux.lisp~	2008-10-08 18:36:19.000000000 +0200
+++ clisp-2.49/modules/bindings/glibc/linux.lisp	2012-12-12 12:50:41.830606483 +0100
@@ -86,7 +86,7 @@
 
 (def-c-type __key_t)            ; int
 
-(c-lines "#include <bits/ipctypes.h>~%")
+(c-lines "#include <sys/ipc.h>~%")
 (def-c-type __ipc_pid_t)        ; ushort
 
 ; --------------------------- <sys/types.h> -----------------------------------
