--- clisp-2.49/modules/bindings/glibc/linux.lisp.orig	2008-10-08 18:36:19.000000000 +0200
+++ clisp-2.49/modules/bindings/glibc/linux.lisp	2017-12-05 15:58:10.508428262 +0100
@@ -67,7 +67,7 @@
 (def-c-type __daddr_t)          ; int
 (def-c-type __caddr_t)          ; c-pointer
 (def-c-type __time_t)           ; long
-(def-c-type __swblk_t)          ; long
+;;(def-c-type __swblk_t)          ; long
 
 (def-c-type __fd_mask ulong)
 (eval-when (load compile eval)
@@ -86,7 +86,7 @@
 
 (def-c-type __key_t)            ; int
 
-(c-lines "#include <bits/ipctypes.h>~%")
+(c-lines "#include <sys/ipc.h>~%")
 (def-c-type __ipc_pid_t)        ; ushort
 
 ; --------------------------- <sys/types.h> -----------------------------------
@@ -293,6 +293,8 @@
 ;; for robust mutexes
 (def-c-const EOWNERDEAD (:documentation "Owner died")) ; 130
 (def-c-const ENOTRECOVERABLE (:documentation "State not recoverable")) ; 131
+(def-c-const ERFKILL (:documentation "Operation not possible due to RF-kill")) ; 132
+(def-c-const EHWPOISON (:documentation "Memory page has hardware error")) ; 133
 
 ; -------------------------- <bits/errno.h> -----------------------------------
 
@@ -648,7 +650,6 @@
 (def-call-out calloc (:arguments (nmemb size_t) (size size_t))
   (:return-type c-pointer))
 (def-call-out free (:arguments (ptr c-pointer)) (:return-type nil))
-(def-call-out cfree (:arguments (ptr c-pointer)) (:return-type nil))
 (def-call-out valloc (:arguments (size size_t)) (:return-type c-pointer))
 
 (def-call-out abort (:arguments) (:return-type nil))
