diff -ur clisp-2.49.orig/src/makemake.in clisp-2.49/src/makemake.in
--- clisp-2.49.orig/src/makemake.in	2019-09-02 21:04:20.246000000 +0200
+++ clisp-2.49/src/makemake.in	2019-09-02 21:06:35.346000000 +0200
@@ -1235,7 +1235,7 @@
   if [ $CROSS = false ] ; then
     case "$XCC_GCC_VERSION" in
       # gcc 2.7 introduced an annoying warning, but gcc 2.8 has a workaround:
-      2.[8-9]* | 3.* | 4.* | egcs-2.*)
+      2.[8-9]* | 3.* | 4.* | 5.* | 6.* | 7.* | 8.* | 9.* | egcs-2.*)
         XCFLAGS=$XCFLAGS' -Wno-sign-compare -Wno-format-nonliteral'
         ;;
       * ) ;;
@@ -1259,7 +1259,7 @@
             # gcc-2.7.2 has a strength-reduction bug (fixed in gcc-2.7.2.1,
             # also fixed through the specs file of some Linux distributions).
             XCFLAGS=$XCFLAGS' -O2 -fexpensive-optimizations -fno-strength-reduce' ;;
-          2.3.[2-9]* | 2.[4-9]* | 3.* | 4.* | egcs-2.*)
+          2.3.[2-9]* | 2.[4-9]* | 3.* | 4.* | 5.* | 6.* | 7.* | 8.* | 9.* | egcs-2.*)
             XCFLAGS=$XCFLAGS' -O2 -fexpensive-optimizations' ;;
           * )
             XCFLAGS=$XCFLAGS' -O' ;;
@@ -1278,7 +1278,7 @@
             # egcs-1.0.3 and egcs-1.1.1 produce incorrect code when compiling
             # lisparit (function UDS_divide_).
             XCFLAGS=$XCFLAGS' -O2 -fno-schedule-insns' ;;
-          2.9* | 3.* | 4.*)
+          2.9* | 3.* | 4.* | 5.* | 6.* | 7.* | 8.* | 9.*)
             # gcc-2.95 produces incorrect code when compiling predtype
             # (function C_coerce).
             XCFLAGS=$XCFLAGS' -O2 -fno-schedule-insns -fno-gcse' ;;
@@ -1313,7 +1313,7 @@
     esac
   fi # !DEBUGGING
   case "$XCC_GCC_VERSION" in
-    3* | 4*)
+    3* | 4* | 5* | 6* | 7* | 8* | 9*)
       # When not optimizing on linux, or with Apple's gcc 4 on MacOS X,
       # or with GCC 4 when "gcc -Os" is used:
       #   C_CODE_ALIGNMENT is wrong. &EVAL-WHEN = 0x806D512
