--- clisp-2.49/src/lispbibl.d~	2016-10-23 08:56:52.000000000 +0200
+++ clisp-2.49/src/lispbibl.d	2016-10-23 09:00:09.471732823 +0200
@@ -203,8 +203,13 @@
       || defined(_M_IX86) || defined(_X86_) /* native Windows */
     #define I80386
   #endif
-  #if defined(__x86_64__) || defined(__amd64__)
-    #define AMD64
+  #if defined(__x86_64__) || defined(__amd64__)
+    #if defined(__ILP32__)
+      #define I80386
+      #define NO_ARI_ASM
+    #else
+      #define AMD64
+    #endif
   #endif
   #if defined(arm) || defined(__arm) || defined(__arm__)
     #define ARM
