diff -ur clisp-2.49.orig/src/makemake.in clisp-2.49/src/makemake.in
--- clisp-2.49.orig/src/makemake.in	2010-07-07 17:15:35.000000000 +0200
+++ clisp-2.49/src/makemake.in	2019-09-02 21:04:20.246000000 +0200
@@ -1138,7 +1138,7 @@
 case "$host_cpu" in
   hppa*) cpu=hppa ;;
   arm*)  cpu=arm  ;;
-  x86_64) cpu=x86_64 ;;
+  x86_64) cpu=i686 ;;
   ia64)   cpu=ia64 ;;
   mips)   cpu=mips ;;
   mips64*) cpu=mips64 ;;
